src/components/templates/product-availability-modal/add-channels-modal-screen.tsx

```diff
@@ -0,0 +1,131 @@
+import { SalesChannel } from "@medusajs/medusa"
+import { useAdminSalesChannels } from "medusa-react"
+import React, { useState } from "react"
+import { usePagination, useRowSelect, useTable } from "react-table"
+import Spinner from "../../atoms/spinner"
+import Button from "../../fundamentals/button"
+import Modal from "../../molecules/modal"
+import { LayeredModalContext } from "../../molecules/modal/layered-modal"
+
+import SalesChannelAvailabilityTable, {
+  useAvailableChannelsModalTableColumns,
+} from "./sales-channel-availability-table"
+
+export const useAddChannelsModalScreen = (
+  selectedRowIds,
+  onAddSalesChannelsToSelected
```

Comment:
suggestion: should we name it `onSelected`? we are in the sales channel context, it seams understandable that we select sales channels

Reply:
I don't like `onSelected` since it's not what is invoked when a row is selected rather it's invoked when 'Add and go back' is clicked. We can call it `onAddAndGoBack` if that is better, but I think `onAddSalesChannelsToSelected` is okay. A different naming could be `onAddToChannelAvailability`, and then to rename everything with the initial modal presented to relate to `channelAvailability`, i.e. `onRemoveFromChannelAvailability` `onSaveChannelAvailability